native-image-tests/local-scala/src/main/scala/com/lightbend/JacksonCheck.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2009-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend
+
+import akka.actor.typed.ActorSystem
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+import akka.serialization.jackson.JsonSerializable
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeName
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+
+object JacksonCheck {
+
+  // from the akka-serialization-jackson docs
+
+  // type hierarchy / ADT
+  // FIXME: Still haven't figured out why @JsonCreator cannot be used here
```

Comment:
This is really strange, I have verified that the constructor and parameter names are available for reflection, and for multi-param constructors it just works with nothing extra, but for some reason just `@JsonCreator` doesn't cut it here and a `@JsonProperty("fieldName")` is required.

I also ran with agent and tried with the generated reflect-config.json (which I expect to add the same or a bit less for the same class) and get the same error so it is something specifically about how Jackson reflectively finds single-param constructors with that annotation.

Reply:
hm, feels like the Jackson Scala module isn't used? Could it be that it's not included in native image and then falls back to regular Java Jackson, which requires the JsonCreator and has the param name problem?

Reply:
Nope, that's not it, the objectMapper lists the same modules for normal JDK run as on native image.

Reply:
I explained elsewhere but it turns out that the Scala Jackson support reads/parses the actual classfiles as resources from the classpath, so needs a resource-config.json entry for graal. The `AkkaJacksonSerializationFeature` now automatically adds that for classes that should be serialisable with Jackson.